// @storybook/design-system/dist/components/tooltip/ListItem.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withLinkWrapper = exports.disabled = exports.wPositionsActiveLongTitle = exports.wPositionsActive = exports.wPositions = exports.activeIcon = exports.defaultIcon = exports.longTitle = exports.defaultStory = exports.loading = exports.all = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ListItem = require("./ListItem");

var _Icon = require("../Icon");

var _StoryLinkWrapper = require("../StoryLinkWrapper");

var _default = {
  title: 'Design System|tooltip/ListItem',
  decorators: [function (storyFn) {
    return _react["default"].createElement("div", {
      style: {
        width: 200,
        border: '1px solid #ddd',
        margin: '3rem'
      }
    }, storyFn());
  }],
  parameters: {
    component: _ListItem.ListItem
  }
};
exports["default"] = _default;

var all = function all() {
  return _react["default"].createElement("div", null, _react["default"].createElement(_ListItem.ListItem, {
    loading: true
  }), _react["default"].createElement(_ListItem.ListItem, {
    title: "Default"
  }), _react["default"].createElement(_ListItem.ListItem, {
    title: "lorem ipsum dolor sit amet consectatur"
  }), _react["default"].createElement(_ListItem.ListItem, {
    title: "Default icon",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  }), _react["default"].createElement(_ListItem.ListItem, {
    left: "left",
    title: "title",
    center: "center",
    right: "right"
  }), _react["default"].createElement(_ListItem.ListItem, {
    active: true,
    left: "left",
    title: "active",
    center: "center",
    right: "right"
  }), _react["default"].createElement(_ListItem.ListItem, {
    appearance: "secondary",
    active: true,
    left: "left",
    title: "secondary active",
    center: "center",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  }), _react["default"].createElement(_ListItem.ListItem, {
    active: true,
    left: "left",
    title: "active icon",
    center: "center",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  }), _react["default"].createElement(_ListItem.ListItem, {
    active: true,
    left: "\u2705",
    title: "lorem ipsum dolor sit amet consectatur",
    center: "center",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  }), _react["default"].createElement(_ListItem.ListItem, {
    disabled: true,
    left: "left",
    title: "disabled",
    center: "center",
    right: "right"
  }));
};

exports.all = all;

var loading = function loading() {
  return _react["default"].createElement(_ListItem.ListItem, {
    loading: true
  });
};

exports.loading = loading;

var defaultStory = function defaultStory() {
  return _react["default"].createElement(_ListItem.ListItem, {
    title: "Default"
  });
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};

var longTitle = function longTitle() {
  return _react["default"].createElement(_ListItem.ListItem, {
    title: "lorem ipsum dolor sit amet consectatur"
  });
};

exports.longTitle = longTitle;
longTitle.story = {
  name: 'long title'
};

var defaultIcon = function defaultIcon() {
  return _react["default"].createElement(_ListItem.ListItem, {
    title: "Default icon",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  });
};

exports.defaultIcon = defaultIcon;
defaultIcon.story = {
  name: 'default icon'
};

var activeIcon = function activeIcon() {
  return _react["default"].createElement(_ListItem.ListItem, {
    active: true,
    title: "active icon",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  });
};

exports.activeIcon = activeIcon;
activeIcon.story = {
  name: 'active icon'
};

var wPositions = function wPositions() {
  return _react["default"].createElement(_ListItem.ListItem, {
    left: "left",
    title: "title",
    center: "center",
    right: "right"
  });
};

exports.wPositions = wPositions;
wPositions.story = {
  name: 'w/positions'
};

var wPositionsActive = function wPositionsActive() {
  return _react["default"].createElement(_ListItem.ListItem, {
    active: true,
    left: "left",
    title: "active",
    center: "center",
    right: "right"
  });
};

exports.wPositionsActive = wPositionsActive;
wPositionsActive.story = {
  name: 'w/positions active'
};

var wPositionsActiveLongTitle = function wPositionsActiveLongTitle() {
  return _react["default"].createElement(_ListItem.ListItem, {
    active: true,
    left: "\u2705",
    title: "lorem ipsum dolor sit amet consectatur",
    center: "center",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    })
  });
};

exports.wPositionsActiveLongTitle = wPositionsActiveLongTitle;
wPositionsActiveLongTitle.story = {
  name: 'w/positions active long title'
};

var disabled = function disabled() {
  return _react["default"].createElement(_ListItem.ListItem, {
    disabled: true,
    left: "left",
    title: "disabled",
    center: "center",
    right: "right"
  });
};

exports.disabled = disabled;

var withLinkWrapper = function withLinkWrapper() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_ListItem.ListItem, {
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    loading: true,
    href: "http://www.google.com"
  }), _react["default"].createElement(_ListItem.ListItem, {
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    title: "Default",
    href: "http://www.google.com"
  }), _react["default"].createElement(_ListItem.ListItem, {
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    title: "lorem ipsum dolor sit amet consectatur",
    to: "http://www.google.com"
  }), _react["default"].createElement(_ListItem.ListItem, {
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    title: "Default icon",
    right: _react["default"].createElement(_Icon.Icon, {
      icon: "eye"
    }),
    to: "http://www.google.com"
  }));
};

exports.withLinkWrapper = withLinkWrapper;
withLinkWrapper.story = {
  name: 'with LinkWrapper'
};
