// @storybook/design-system/dist/components/tooltip/TooltipLinkList.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TooltipLinkList = TooltipLinkList;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ListItem = require("./ListItem");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  min-width: 180px;\n  border-radius: 4px;\n  overflow: hidden;\n  margin: 0;\n  padding: 0;\n  list-style: none;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var List = _styledComponents["default"].ul(_templateObject());

function TooltipLinkList(_ref) {
  var links = _ref.links,
      LinkWrapper = _ref.LinkWrapper;
  return _react["default"].createElement(List, null, links.map(function (_ref2, index) {
    var title = _ref2.title,
        onClick = _ref2.onClick,
        active = _ref2.active,
        rest = (0, _objectWithoutProperties2["default"])(_ref2, ["title", "onClick", "active"]);
    return _react["default"].createElement(_ListItem.ListItem
    /* eslint-disable react/no-array-index-key */
    , (0, _extends2["default"])({
      key: index,
      title: title,
      onClick: onClick,
      active: active,
      LinkWrapper: LinkWrapper || null
    }, rest));
  }));
}

TooltipLinkList.propTypes = {
  links: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    title: _propTypes["default"].node.isRequired,
    onClick: _propTypes["default"].func,
    active: _propTypes["default"].bool
  }).isRequired).isRequired,
  LinkWrapper: _propTypes["default"].func
};
TooltipLinkList.defaultProps = {
  LinkWrapper: _ListItem.ListItem.defaultProps.LinkWrapper
};
