// @storybook/design-system/dist/components/tooltip/TooltipNote.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultStory = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _WithTooltip = _interopRequireDefault(require("./WithTooltip"));

var _TooltipNote = require("./TooltipNote");

var _default = {
  title: 'Design System|tooltip/TooltipNote',
  decorators: [function (storyFn) {
    return _react["default"].createElement("div", {
      style: {
        height: '300px'
      }
    }, _react["default"].createElement(_WithTooltip["default"], {
      hasChrome: false,
      placement: "top",
      trigger: "click",
      startOpen: true,
      tooltip: storyFn()
    }, _react["default"].createElement("div", null, "Tooltip")));
  }],
  parameters: {
    component: _TooltipNote.TooltipNote
  }
};
exports["default"] = _default;

var defaultStory = function defaultStory() {
  return _react["default"].createElement(_TooltipNote.TooltipNote, {
    note: "Lorem ipsum dolor"
  });
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};
