// @storybook/design-system/dist/components/tooltip/TooltipMessage.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.minimalMessage = exports.withLinks = exports.withLink = exports.defaultStory = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _WithTooltip = _interopRequireDefault(require("./WithTooltip"));

var _TooltipMessage = require("./TooltipMessage");

var _default = {
  title: 'Design System|tooltip/TooltipMessage',
  decorators: [function (storyFn) {
    return _react["default"].createElement("div", {
      style: {
        height: '300px'
      }
    }, _react["default"].createElement(_WithTooltip["default"], {
      placement: "top",
      trigger: "click",
      startOpen: true,
      tooltip: storyFn()
    }, _react["default"].createElement("div", null, "Tooltip")));
  }],
  parameters: {
    component: _TooltipMessage.TooltipMessage
  }
};
exports["default"] = _default;

var defaultStory = function defaultStory() {
  return _react["default"].createElement(_TooltipMessage.TooltipMessage, {
    title: "Lorem ipsum dolor sit",
    desc: "Amet consectatur vestibulum concet durum politu coret weirom"
  });
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};

var withLink = function withLink() {
  return _react["default"].createElement(_TooltipMessage.TooltipMessage, {
    title: "Lorem ipsum dolor sit",
    desc: "Amet consectatur vestibulum concet durum politu coret weirom",
    links: [{
      title: 'Continue',
      href: 'test'
    }]
  });
};

exports.withLink = withLink;
withLink.story = {
  name: 'with link'
};

var withLinks = function withLinks() {
  return _react["default"].createElement(_TooltipMessage.TooltipMessage, {
    title: "Lorem ipsum dolor sit",
    desc: "Amet consectatur vestibulum concet durum politu coret weirom",
    links: [{
      title: 'Get more tips',
      href: 'test'
    }, {
      title: 'Done',
      href: 'test'
    }]
  });
};

exports.withLinks = withLinks;
withLinks.story = {
  name: 'with links'
};

var minimalMessage = function minimalMessage() {
  return _react["default"].createElement(_TooltipMessage.TooltipMessage, {
    desc: "Amet consectatur vestibulum concet durum politu coret weirom"
  });
};

exports.minimalMessage = minimalMessage;
minimalMessage.story = {
  name: 'minimal message'
};
