// @storybook/design-system/dist/components/tooltip/TooltipLinkList.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.linksAndCallback = exports.story0 = exports["default"] = exports.links = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _WithTooltip = _interopRequireDefault(require("./WithTooltip"));

var _TooltipLinkList = require("./TooltipLinkList");

var _StoryLinkWrapper = require("../StoryLinkWrapper");

var links = [{
  title: 'Link',
  href: 'https://learnstorybook.com'
}, {
  title: 'Link',
  href: 'https://learnstorybook.com'
}, {
  title: 'callback',
  onClick: (0, _addonActions.action)('onClick')
}];
exports.links = links;
var _default = {
  title: 'Design System|tooltip/TooltipLinkList',
  component: _TooltipLinkList.TooltipLinkList,
  decorators: [function (storyFn) {
    return _react["default"].createElement("div", {
      style: {
        height: '300px'
      }
    }, _react["default"].createElement(_WithTooltip["default"], {
      placement: "top",
      trigger: "click",
      startOpen: true,
      tooltip: storyFn()
    }, _react["default"].createElement("div", null, "Tooltip")));
  }],
  excludeStories: ['links']
};
exports["default"] = _default;

var story0 = function story0() {
  return _react["default"].createElement(_TooltipLinkList.TooltipLinkList, {
    links: links.slice(0, 2),
    LinkWrapper: null
  });
};

exports.story0 = story0;
story0.story = {
  name: 'links'
};

var linksAndCallback = function linksAndCallback() {
  return _react["default"].createElement(_TooltipLinkList.TooltipLinkList, {
    links: links,
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper
  });
};

exports.linksAndCallback = linksAndCallback;
linksAndCallback.story = {
  name: 'links and callback'
};
