// @storybook/design-system/dist/components/tooltip/Tooltip.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.noChrome = exports.basicDefaultRight = exports.basicDefaultLeft = exports.basicDefaultBottom = exports.basicDefault = exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Tooltip = require("./Tooltip");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 100px;\n  height: 100px;\n  font-size: 16px;\n  text-align: center;\n  line-height: 100px;\n  background: #eee;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

// Popper would position the tooltip absolutely. We just need to make sure we are pos:rel
var mockPopperProps = {
  style: {
    position: 'relative',
    top: '20px',
    left: '20px'
  }
};

var Content = _styledComponents["default"].div(_templateObject());

var _default = {
  title: 'Design System|tooltip/Tooltip',
  parameters: {
    component: _Tooltip.Tooltip
  }
};
exports["default"] = _default;

var basicDefault = function basicDefault() {
  return _react["default"].createElement(_Tooltip.Tooltip, (0, _extends2["default"])({
    hasChrome: true
  }, mockPopperProps), _react["default"].createElement(Content, null, "Text"));
};

exports.basicDefault = basicDefault;
basicDefault.story = {
  name: 'basic, default'
};

var basicDefaultBottom = function basicDefaultBottom() {
  return _react["default"].createElement(_Tooltip.Tooltip, (0, _extends2["default"])({
    hasChrome: true,
    placement: "bottom"
  }, mockPopperProps), _react["default"].createElement(Content, null, "Text"));
};

exports.basicDefaultBottom = basicDefaultBottom;
basicDefaultBottom.story = {
  name: 'basic, default, bottom'
};

var basicDefaultLeft = function basicDefaultLeft() {
  return _react["default"].createElement(_Tooltip.Tooltip, (0, _extends2["default"])({
    hasChrome: true,
    placement: "left"
  }, mockPopperProps), _react["default"].createElement(Content, null, "Text"));
};

exports.basicDefaultLeft = basicDefaultLeft;
basicDefaultLeft.story = {
  name: 'basic, default, left'
};

var basicDefaultRight = function basicDefaultRight() {
  return _react["default"].createElement(_Tooltip.Tooltip, (0, _extends2["default"])({
    placement: "right"
  }, mockPopperProps), _react["default"].createElement(Content, null, "Text"));
};

exports.basicDefaultRight = basicDefaultRight;
basicDefaultRight.story = {
  name: 'basic, default, right'
};

var noChrome = function noChrome() {
  return _react["default"].createElement(_Tooltip.Tooltip, mockPopperProps, _react["default"].createElement(Content, null, "Text"));
};

exports.noChrome = noChrome;
noChrome.story = {
  name: 'no chrome'
};
