// @storybook/design-system/dist/components/tooltip/WithTooltip.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.noChrome = exports.simpleClickCloseOnClick = exports.simpleClickStartOpen = exports.simpleClickAsSvg = exports.simpleClick = exports.simpleHoverFunctional = exports.simpleHoverAsSvg = exports.simpleHover = exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _TooltipMessage = require("./TooltipMessage");

var _WithTooltip = _interopRequireDefault(require("./WithTooltip"));

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  fill: fill: rgb(255, 0, 255);\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 200px;\n  height: 100px;\n  background-color: red;\n  color: white;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  height: 100px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 500px;\n  height: 500px;\n  background: pink;\n  position: relative;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  height: 300px;\n  background: purple;\n  overflow-y: scroll;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ViewPort = _styledComponents["default"].div(_templateObject());

var BackgroundBox = _styledComponents["default"].div(_templateObject2());

var Spacer = _styledComponents["default"].div(_templateObject3());

var Trigger = _styledComponents["default"].div(_templateObject4());

var Circle = _styledComponents["default"].circle(_templateObject5());

var Tooltip = function Tooltip(_ref) {
  var onHide = _ref.onHide;
  return _react["default"].createElement(_TooltipMessage.TooltipMessage, {
    title: "Lorem ipsum dolor sit",
    desc: "Amet consectatur vestibulum concet durum politu coret weirom",
    links: [{
      title: 'Continue',
      onClick: onHide
    }]
  });
};

Tooltip.propTypes = {
  onHide: _propTypes["default"].func
};
Tooltip.defaultProps = {
  onHide: null
};
var _default = {
  title: 'Design System|tooltip/WithTooltip',
  decorators: [function (storyFn) {
    return _react["default"].createElement(ViewPort, null, _react["default"].createElement(BackgroundBox, null, _react["default"].createElement(Spacer, null), storyFn()));
  }],
  parameters: {
    component: _WithTooltip["default"]
  }
};
exports["default"] = _default;

var simpleHover = function simpleHover() {
  return _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "hover",
    tooltip: _react["default"].createElement(Tooltip, null)
  }, _react["default"].createElement(Trigger, null, "Hover me!"));
};

exports.simpleHover = simpleHover;
simpleHover.story = {
  name: 'simple hover'
};

var simpleHoverAsSvg = function simpleHoverAsSvg() {
  return _react["default"].createElement("svg", null, _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "hover",
    tooltip: _react["default"].createElement(Tooltip, null),
    tagName: "g",
    "aria-label": "Empty box to hover for tooltip"
  }, _react["default"].createElement(Circle, {
    cx: "150",
    cy: "50",
    r: "50"
  })));
};

exports.simpleHoverAsSvg = simpleHoverAsSvg;
simpleHoverAsSvg.story = {
  name: 'simple hover, as svg'
};

var simpleHoverFunctional = function simpleHoverFunctional() {
  return _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "hover",
    tooltip: Tooltip
  }, _react["default"].createElement(Trigger, null, "Hover me!"));
};

exports.simpleHoverFunctional = simpleHoverFunctional;
simpleHoverFunctional.story = {
  name: 'simple hover, functional'
};

var simpleClick = function simpleClick() {
  return _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "click",
    tooltip: _react["default"].createElement(Tooltip, null)
  }, _react["default"].createElement(Trigger, null, "Click me!"));
};

exports.simpleClick = simpleClick;
simpleClick.story = {
  name: 'simple click'
};

var simpleClickAsSvg = function simpleClickAsSvg() {
  return _react["default"].createElement("svg", null, _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "click",
    tooltip: _react["default"].createElement(Tooltip, null),
    tagName: "g",
    "aria-label": "Empty box to click for tooltip"
  }, _react["default"].createElement(Circle, {
    cx: "150",
    cy: "50",
    r: "50"
  })));
};

exports.simpleClickAsSvg = simpleClickAsSvg;
simpleClickAsSvg.story = {
  name: 'simple click, as svg'
};

var simpleClickStartOpen = function simpleClickStartOpen() {
  return _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "click",
    startOpen: true,
    tooltip: _react["default"].createElement(Tooltip, null)
  }, _react["default"].createElement(Trigger, null, "Click me!"));
};

exports.simpleClickStartOpen = simpleClickStartOpen;
simpleClickStartOpen.story = {
  name: 'simple click start open'
};

var simpleClickCloseOnClick = function simpleClickCloseOnClick() {
  return _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "click",
    closeOnClick: true,
    tooltip: _react["default"].createElement(Tooltip, null)
  }, _react["default"].createElement(Trigger, null, "Click me!"));
};

exports.simpleClickCloseOnClick = simpleClickCloseOnClick;
simpleClickCloseOnClick.story = {
  name: 'simple click closeOnClick'
};

var noChrome = function noChrome() {
  return _react["default"].createElement(_WithTooltip["default"], {
    placement: "top",
    trigger: "click",
    hasChrome: false,
    tooltip: _react["default"].createElement(Tooltip, null)
  }, _react["default"].createElement(Trigger, null, "Click me!"));
};

exports.noChrome = noChrome;
noChrome.story = {
  name: 'no chrome'
};
