// @storybook/design-system/dist/components/AvatarList.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.empty = exports.loading = exports.smallSize = exports.bigUserCount = exports.ellipsized = exports["short"] = exports["default"] = exports.users = void 0;

var _react = _interopRequireDefault(require("react"));

var _AvatarList = require("./AvatarList");

var users = [{
  id: '1',
  name: 'Dominic Nguyen',
  avatarUrl: 'https://avatars2.githubusercontent.com/u/263385'
}, {
  id: '2',
  name: 'Tom Coleman',
  avatarUrl: 'https://avatars2.githubusercontent.com/u/132554'
}, {
  id: '3',
  name: 'Zoltan Olah',
  avatarUrl: 'https://avatars0.githubusercontent.com/u/81672'
}, {
  id: '4',
  name: 'Tim Hingston',
  avatarUrl: 'https://avatars3.githubusercontent.com/u/1831709'
}];
exports.users = users;
var _default = {
  title: 'Design System|AvatarList',
  component: _AvatarList.AvatarList,
  excludeStories: ['users']
};
exports["default"] = _default;

var _short = function _short() {
  return _react["default"].createElement(_AvatarList.AvatarList, {
    users: users.slice(0, 2)
  });
};

exports["short"] = _short;

var ellipsized = function ellipsized() {
  return _react["default"].createElement(_AvatarList.AvatarList, {
    users: users
  });
};

exports.ellipsized = ellipsized;

var bigUserCount = function bigUserCount() {
  return _react["default"].createElement(_AvatarList.AvatarList, {
    users: users,
    userCount: 100
  });
};

exports.bigUserCount = bigUserCount;
bigUserCount.story = {
  name: 'big userCount'
};

var smallSize = function smallSize() {
  return _react["default"].createElement(_AvatarList.AvatarList, {
    users: users,
    userCount: 100,
    size: "small"
  });
};

exports.smallSize = smallSize;
smallSize.story = {
  name: 'small size'
};

var loading = function loading() {
  return _react["default"].createElement(_AvatarList.AvatarList, {
    loading: true
  });
};

exports.loading = loading;

var empty = function empty() {
  return _react["default"].createElement(_AvatarList.AvatarList, {
    users: []
  });
};

exports.empty = empty;
