// @storybook/design-system/dist/components/Avatar.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tiny = exports.small = exports.medium = exports.large = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Avatar = require("./Avatar");

var _default = {
  title: 'Design System|Avatar',
  parameters: {
    component: _Avatar.Avatar
  }
};
exports["default"] = _default;

var large = function large() {
  return _react["default"].createElement("div", null, _react["default"].createElement(_Avatar.Avatar, {
    loading: true,
    size: "large"
  }), _react["default"].createElement(_Avatar.Avatar, {
    size: "large",
    username: "Tom Coleman"
  }), _react["default"].createElement(_Avatar.Avatar, {
    size: "large",
    username: "Tom Coleman",
    src: "https://avatars2.githubusercontent.com/u/132554"
  }));
};

exports.large = large;

var medium = function medium() {
  return _react["default"].createElement("div", null, _react["default"].createElement(_Avatar.Avatar, {
    loading: true
  }), _react["default"].createElement(_Avatar.Avatar, {
    username: "Tom Coleman"
  }), _react["default"].createElement(_Avatar.Avatar, {
    username: "Tom Coleman",
    src: "https://avatars2.githubusercontent.com/u/132554"
  }));
};

exports.medium = medium;

var small = function small() {
  return _react["default"].createElement("div", null, _react["default"].createElement(_Avatar.Avatar, {
    loading: true,
    size: "small"
  }), _react["default"].createElement(_Avatar.Avatar, {
    size: "small",
    username: "Dominic Nguyen"
  }), _react["default"].createElement(_Avatar.Avatar, {
    size: "small",
    username: "Dominic Nguyen",
    src: "https://avatars2.githubusercontent.com/u/263385"
  }));
};

exports.small = small;

var tiny = function tiny() {
  return _react["default"].createElement("div", null, _react["default"].createElement(_Avatar.Avatar, {
    loading: true,
    size: "tiny"
  }), _react["default"].createElement(_Avatar.Avatar, {
    size: "tiny",
    username: "Dominic Nguyen"
  }), _react["default"].createElement(_Avatar.Avatar, {
    size: "tiny",
    username: "Dominic Nguyen",
    src: "https://avatars2.githubusercontent.com/u/263385"
  }));
};

exports.tiny = tiny;
