// @storybook/design-system/dist/components/AvatarList.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AvatarList = AvatarList;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Avatar = require("./Avatar");

var _WithTooltip = _interopRequireDefault(require("./tooltip/WithTooltip"));

var _TooltipNote = require("./tooltip/TooltipNote");

var _styles = require("./shared/styles");

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n  flex-wrap: nowrap;\n  flex-direction: row;\n  align-items: center;\n  justify-content: flex-end;\n  vertical-align: top;\n  margin: 0;\n  padding: 0;\n  list-style: none;\n\n  ", " {\n    position: relative;\n\n    &:not(:first-child) {\n      margin-left: -6px;\n    }\n    &:nth-child(1) {\n      z-index: 3;\n    }\n    &:nth-child(2) {\n      z-index: 2;\n    }\n    &:nth-child(3) {\n      z-index: 1;\n    }\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n  font-size: ", "px;\n  color: ", ";\n  margin-left: 6px;\n  white-space: nowrap;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  box-shadow: ", " 0 0 0 2px;\n  display: block;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var UserAvatar = (0, _styledComponents["default"])(_Avatar.Avatar)(_templateObject(), _styles.color.lightest);
var UserTooltipWrapper = (0, _styledComponents["default"])(_WithTooltip["default"])(_templateObject2());

var UserEllipses = _styledComponents["default"].li(_templateObject3(), _styles.typography.size.s1, _styles.color.mediumdark);

var User = _styledComponents["default"].li(_templateObject4());

var Users = _styledComponents["default"].ul(_templateObject5(), User); // Either pass the full list of users, or a userCount if known


function AvatarList(_ref) {
  var loading = _ref.loading,
      users = _ref.users,
      userCount = _ref.userCount,
      size = _ref.size,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["loading", "users", "userCount", "size"]);
  var count = userCount || users.length;
  return _react["default"].createElement(Users, (0, _extends2["default"])({
    "aria-label": "users"
  }, props), users.slice(0, 3).map(function (_ref2) {
    var id = _ref2.id,
        name = _ref2.name,
        avatarUrl = _ref2.avatarUrl;
    return _react["default"].createElement(User, {
      key: id
    }, _react["default"].createElement(UserTooltipWrapper, {
      hasChrome: false,
      placement: "bottom",
      mode: "hover",
      tooltip: _react["default"].createElement(_TooltipNote.TooltipNote, {
        note: name
      })
    }, _react["default"].createElement(UserAvatar, {
      size: size,
      username: name,
      src: avatarUrl,
      loading: loading
    })));
  }), count > 3 && _react["default"].createElement(UserEllipses, {
    "aria-label": "".concat(count - 3, " more user(s)")
  }, " + ", count - 3, " "));
}

AvatarList.propTypes = {
  loading: _propTypes["default"].bool,
  users: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    id: _propTypes["default"].string.isRequired,
    name: _propTypes["default"].string,
    avatarUrl: _propTypes["default"].string
  })),
  userCount: _propTypes["default"].number,
  size: _propTypes["default"].oneOf(Object.keys(_Avatar.sizes))
};
AvatarList.defaultProps = {
  loading: false,
  users: [{
    id: 'loading',
    avatarUrl: null,
    name: 'loading'
  }, {
    id: 'loading2',
    avatarUrl: null,
    name: 'loading'
  }, {
    id: 'loading3',
    avatarUrl: null,
    name: 'loading'
  }],
  userCount: null,
  size: 'medium'
};
