// @storybook/design-system/dist/components/Badge.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withIcon = exports.all = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Badge = require("./Badge");

var _Icon = require("./Icon");

var _default = {
  title: 'Design System|Badge',
  component: _Badge.Badge
};
exports["default"] = _default;

var all = function all() {
  return _react["default"].createElement("div", null, _react["default"].createElement(_Badge.Badge, {
    status: "positive"
  }, "Positive"), _react["default"].createElement(_Badge.Badge, {
    status: "negative"
  }, "Negative"), _react["default"].createElement(_Badge.Badge, {
    status: "neutral"
  }, "Neutral"), _react["default"].createElement(_Badge.Badge, {
    status: "error"
  }, "Error"), _react["default"].createElement(_Badge.Badge, {
    status: "warning"
  }, "Warning"));
};

exports.all = all;

var withIcon = function withIcon() {
  return _react["default"].createElement(_Badge.Badge, {
    status: "warning"
  }, _react["default"].createElement(_Icon.Icon, {
    icon: "check",
    inline: true
  }), "with icon");
};

exports.withIcon = withIcon;
withIcon.story = {
  name: 'with icon'
};
