// @storybook/design-system/dist/components/Checkbox.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.checked = exports.unchecked = exports.allCheckboxes = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Checkbox = require("./Checkbox");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'Design System|forms/Checkbox',
  component: _Checkbox.Checkbox
};
exports["default"] = _default;

var allCheckboxes = function allCheckboxes() {
  return _react["default"].createElement("form", null, _react["default"].createElement(_Checkbox.Checkbox, {
    id: "Unchecked",
    label: "Cats",
    hideLabel: true,
    onChange: onChange
  }), _react["default"].createElement(_Checkbox.Checkbox, {
    id: "Checked",
    label: "Cats",
    hideLabel: true,
    checked: true,
    onChange: onChange
  }), _react["default"].createElement(_Checkbox.Checkbox, {
    id: "With-label-and-error",
    label: "Cats",
    onChange: onChange,
    error: "There's a snake in my boots"
  }), _react["default"].createElement(_Checkbox.Checkbox, {
    id: "With-label",
    label: "Cats",
    onChange: onChange
  }), _react["default"].createElement(_Checkbox.Checkbox, {
    appearance: "secondary",
    id: "With-label",
    label: "Secondary",
    checked: true,
    onChange: onChange
  }), _react["default"].createElement(_Checkbox.Checkbox, {
    appearance: "secondary",
    id: "With-label",
    label: "Secondary",
    onChange: onChange
  }));
};

exports.allCheckboxes = allCheckboxes;
allCheckboxes.story = {
  name: 'all checkboxes'
};

var unchecked = function unchecked() {
  return _react["default"].createElement(_Checkbox.Checkbox, {
    id: "Unchecked",
    label: "Cats",
    hideLabel: true,
    onChange: onChange
  });
};

exports.unchecked = unchecked;

var checked = function checked() {
  return _react["default"].createElement(_Checkbox.Checkbox, {
    id: "Checked",
    label: "Cats",
    hideLabel: true,
    checked: true,
    onChange: onChange
  });
};

exports.checked = checked;
