// @storybook/design-system/dist/components/Highlight.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Highlight = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _prismjs = _interopRequireDefault(require("prismjs"));

var _index = _interopRequireDefault(require("prismjs/components/index"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("./shared/styles");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  /*\n    Line below from: prismjs/themes/prism.css, with removal of overly specific\n    selectors (primarily [class*=language-]) and selectors that collide with our\n    global styling (primarly code & pre selectors).\n  */\n  .token.cdata,.token.comment,.token.doctype,.token.prolog{color:#708090}.token.punctuation{color:#999}.namespace{opacity:.7}.token.boolean,.token.constant,.token.deleted,.token.number,.token.property,.token.symbol,.token.tag{color:#905}.token.attr-name,.token.builtin,.token.char,.token.inserted,.token.selector,.token.string{color:#690}.language-css .token.string,.style .token.string,.token.entity,.token.operator,.token.url{color:#a67f59;background:hsla(0,0%,100%,.5)}.token.atrule,.token.attr-value,.token.keyword{color:#07a}.token.class-name,.token.function{color:#dd4a68}.token.important,.token.regex,.token.variable{color:#e90}.token.bold,.token.important{font-weight:700}.token.italic{font-style:italic}.token.entity{cursor:help}\n\n  code, pre {\n    color: ", ";\n  }\n\n  code {\n    white-space: pre;\n  }\n\n  pre {\n    padding: 11px 1rem;\n    margin: 1rem 0;\n    background: ", ";\n    overflow: auto;\n  }\n\n  .language-bash .token.operator,\n  .language-bash .token.function {\n    color: ", ";\n    background: none;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var languages = ['bash', 'javascript', 'typescript', 'json', 'css', 'yaml'];
(0, _index["default"])(languages); // Prism theme copied from 'prismjs/themes/prism.css.' -- without Webpack, the CSS
// cannot be imported easily and any app which pulls in the design system will
// need to handle the CSS loading itself. Therefore, it is easiest to just copy
// the theme over.
// prettier-ignore

var HighlightBlock = _styledComponents["default"].div(_templateObject(), _styles.color.darkest, _styles.color.lighter, _styles.color.darkest);

var Highlight =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2["default"])(Highlight, _React$Component);

  function Highlight() {
    (0, _classCallCheck2["default"])(this, Highlight);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(Highlight).apply(this, arguments));
  }

  (0, _createClass2["default"])(Highlight, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.highlightCode();
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      this.highlightCode();
    }
  }, {
    key: "highlightCode",
    value: function highlightCode() {
      var domNode = _reactDom["default"].findDOMNode(this); // eslint-disable-line


      _prismjs["default"].highlightAllUnder(domNode);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          language = _this$props.language,
          rest = (0, _objectWithoutProperties2["default"])(_this$props, ["children", "language"]);

      var codeBlock = _react["default"].createElement("div", {
        dangerouslySetInnerHTML: {
          __html: children
        }
      });

      return _react["default"].createElement(HighlightBlock, rest, language ? _react["default"].createElement("pre", {
        className: "language-".concat(language)
      }, _react["default"].createElement("code", {
        className: "language-".concat(language)
      }, codeBlock)) : codeBlock);
    }
  }]);
  return Highlight;
}(_react["default"].Component);

exports.Highlight = Highlight;
Highlight.propTypes = {
  children: _propTypes["default"].node.isRequired,
  language: _propTypes["default"].oneOf(languages)
};
Highlight.defaultProps = {
  language: null
};
