// @storybook/design-system/dist/components/Highlight.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.customStyling = exports.wrapping = exports.yaml = exports.json = exports.css = exports.typescript = exports.javascript = exports.bash = exports.allLanguages = exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Highlight = require("./Highlight");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  code,\n  pre {\n    font-size: 16px;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  max-width: 300px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var bashCode = "# Highlight bash:\nnpx install some-package-name\ncd some-package-name";
var bashCodeWithWrappers = "<pre class=\"language-bash\"><code class=\"language-bash\">".concat(bashCode, "</code></pre>");
var javascriptCode = "// Highlight JavaScript:\nimport React from 'react'\n\nconst MyComponent = () => (\n  &#x3C;div>My component renders all the things&#x3C;/div>\n)\n\nexport default MyComponent\n";
var javascriptCodeWithWrappers = "<pre class=\"language-javascript\"><code class=\"language-javascript\">".concat(javascriptCode, "</code></pre>");
var typescriptCode = "// Highlight Typescript:\nimport React from 'react'\n\ninterface InterfaceMyComponentProps {\n  isCool: boolean;\n}\n\nconst MyComponent: React.SFC&#x3C;InterfaceMyComponentProps> = ({ isCool }) => (\n  &#x3C;div>My component isCool: {isCool}&#x3C;/div>\n)\n\nexport default MyComponent\n";
var typescriptCodeWithWrappers = "<pre class=\"language-typescript\"><code class=\"language-typescript\">".concat(typescriptCode, "</code></pre>");
var cssCode = "/* Highlight CSS: */\n.someClass {\n  position: relative;\n}\n";
var cssCodeWithWrappers = "<pre class=\"language-css\"><code class=\"language-css\">".concat(cssCode, "</code></pre>");
var jsonCode = "{\n  \"name\": \"@yourorg/package\",\n  \"version\": \"0.0.1\",\n  \"description\": \"A sweet package\",\n  \"repository\": {\n    \"type\": \"git\",\n    \"url\": \"https://github.com/yourorg/package.git\"\n  }\n}";
var jsonCodeWithWrappers = "<pre class=\"language-json\"><code class=\"language-json\">".concat(jsonCode, "</code></pre>");
var yamlCode = "version: 2\njobs:\n  build:\n    docker:\n      - image: circleci/node:8.10.0\n\n    working_directory: ~/repo\n\n    steps:\n      - checkout\n";
var yamlCodeWithWrappers = "<pre class=\"language-yaml\"><code class=\"language-yaml\">".concat(yamlCode, "</code></pre>");
var _default = {
  title: 'Design System|Highlight',
  component: _Highlight.Highlight
};
exports["default"] = _default;

var allLanguages = function allLanguages() {
  return _react["default"].createElement(_Highlight.Highlight, null, [bashCodeWithWrappers, javascriptCodeWithWrappers, typescriptCodeWithWrappers, cssCodeWithWrappers, jsonCodeWithWrappers, yamlCodeWithWrappers].join(''));
};

exports.allLanguages = allLanguages;
allLanguages.story = {
  name: 'all languages'
};

var bash = function bash() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("strong", null, "Autoformat"), _react["default"].createElement(_Highlight.Highlight, {
    language: "bash"
  }, bashCode), _react["default"].createElement("strong", null, "Pre-formatted"), _react["default"].createElement(_Highlight.Highlight, null, bashCodeWithWrappers));
};

exports.bash = bash;

var javascript = function javascript() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("strong", null, "Autoformat"), _react["default"].createElement(_Highlight.Highlight, {
    language: "javascript"
  }, javascriptCode), _react["default"].createElement("strong", null, "Pre-formatted"), _react["default"].createElement(_Highlight.Highlight, null, javascriptCodeWithWrappers));
};

exports.javascript = javascript;

var typescript = function typescript() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("strong", null, "Autoformat"), _react["default"].createElement(_Highlight.Highlight, {
    language: "typescript"
  }, typescriptCode), _react["default"].createElement("strong", null, "Pre-formatted"), _react["default"].createElement(_Highlight.Highlight, null, typescriptCodeWithWrappers));
};

exports.typescript = typescript;

var css = function css() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("strong", null, "Autoformat"), _react["default"].createElement(_Highlight.Highlight, {
    language: "css"
  }, cssCode), _react["default"].createElement("strong", null, "Pre-formatted"), _react["default"].createElement(_Highlight.Highlight, null, cssCodeWithWrappers));
};

exports.css = css;

var json = function json() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("strong", null, "Autoformat"), _react["default"].createElement(_Highlight.Highlight, {
    language: "json"
  }, jsonCode), _react["default"].createElement("strong", null, "Pre-formatted"), _react["default"].createElement(_Highlight.Highlight, null, jsonCodeWithWrappers));
};

exports.json = json;

var yaml = function yaml() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("strong", null, "Autoformat"), _react["default"].createElement(_Highlight.Highlight, {
    language: "yaml"
  }, yamlCode), _react["default"].createElement("strong", null, "Pre-formatted"), _react["default"].createElement(_Highlight.Highlight, null, yamlCodeWithWrappers));
};

exports.yaml = yaml;

var SmallContainer = _styledComponents["default"].div(_templateObject());

var wrapping = function wrapping() {
  return _react["default"].createElement(SmallContainer, null, _react["default"].createElement(_Highlight.Highlight, {
    language: "javascript"
  }, javascriptCode));
};

exports.wrapping = wrapping;
var StyledHighlight = (0, _styledComponents["default"])(_Highlight.Highlight)(_templateObject2());

var customStyling = function customStyling() {
  return _react["default"].createElement(StyledHighlight, {
    language: "json"
  }, jsonCode);
};

exports.customStyling = customStyling;
