// @storybook/design-system/dist/components/Input.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.code = exports.pill = exports.tertiary = exports.secondary = exports.defaultStory = exports.allInputs = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Input = require("./Input");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'Design System|forms/Input',
  component: _Input.Input
};
exports["default"] = _default;

var allInputs = function allInputs() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Input.Input, {
    id: "Default",
    value: "Default",
    label: "Email",
    hideLabel: true,
    icon: "email",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Secondary",
    value: "Secondary",
    label: "Email",
    hideLabel: true,
    icon: "email",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Secondary-with-label",
    value: "Secondary",
    label: "Label secondary",
    icon: "email",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Tertiary",
    value: "Tertiary",
    label: "Email",
    hideLabel: true,
    icon: "email",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Pill",
    value: "Pill",
    label: "Search",
    hideLabel: true,
    icon: "search",
    appearance: "pill",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Code",
    value: "Code",
    label: "Code",
    hideLabel: true,
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Code-horizontal",
    value: "Code",
    appearance: "code",
    orientation: "horizontal",
    onChange: onChange,
    label: "horizontal"
  }), _react["default"].createElement(_Input.Input, {
    id: "Code-secondary-horizontal",
    value: "Code",
    appearance: "secondary",
    orientation: "horizontal",
    onChange: onChange,
    label: "horizontal"
  }));
};

exports.allInputs = allInputs;
allInputs.story = {
  name: 'all inputs'
};

var defaultStory = function defaultStory() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Input.Input, {
    id: "Placeholder",
    label: "Input with placeholder",
    hideLabel: true,
    placeholder: "Placeholder",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "With-value",
    value: "With value",
    label: "Input with value",
    hideLabel: true,
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Disabled",
    value: "Disabled",
    label: "Disabled input",
    hideLabel: true,
    disabled: true,
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Icon",
    value: "Icon",
    label: "Input with icon",
    hideLabel: true,
    icon: "email",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Error",
    label: "Input with error",
    hideLabel: true,
    placeholder: "Error",
    error: "There's a snake in my boots",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Error with icon",
    label: "Input with error and icon",
    hideLabel: true,
    placeholder: "Error with icon",
    icon: "email",
    error: "There's a snake in my boots",
    onChange: onChange
  }));
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};

var secondary = function secondary() {
  return _react["default"].createElement("form", {
    style: {
      background: '#fff',
      padding: '3em'
    }
  }, _react["default"].createElement(_Input.Input, {
    id: "Placeholder",
    label: "Input with placeholder",
    hideLabel: true,
    placeholder: "Placeholder",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "With-value",
    value: "With value",
    label: "Input with value",
    hideLabel: true,
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Disabled",
    value: "Disabled",
    label: "Disabled input",
    hideLabel: true,
    disabled: true,
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Icon",
    value: "Icon",
    label: "Input with icon",
    hideLabel: true,
    icon: "email",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Error",
    label: "Input with error",
    hideLabel: true,
    placeholder: "Error",
    error: "There's a snake in my boots",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "With-label",
    value: "With value",
    label: "Label secondary",
    appearance: "secondary",
    onChange: onChange
  }));
};

exports.secondary = secondary;

var tertiary = function tertiary() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Input.Input, {
    id: "Placeholder",
    label: "Input with placeholder",
    hideLabel: true,
    placeholder: "Placeholder",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "With-value",
    value: "With value",
    label: "Input with value",
    hideLabel: true,
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Disabled",
    value: "Disabled",
    label: "Disabled input",
    hideLabel: true,
    disabled: true,
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Icon",
    value: "Icon",
    label: "Input with icon",
    hideLabel: true,
    icon: "email",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Error",
    label: "Input with error",
    hideLabel: true,
    placeholder: "Error",
    error: "There's a snake in my boots",
    appearance: "tertiary",
    onChange: onChange
  }));
};

exports.tertiary = tertiary;

var pill = function pill() {
  return _react["default"].createElement(_Input.Input, {
    id: "Pill",
    value: "Pill",
    label: "Search",
    hideLabel: true,
    icon: "search",
    appearance: "pill",
    onChange: onChange
  });
};

exports.pill = pill;

var code = function code() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Input.Input, {
    id: "Code-placeholder",
    placeholder: "Code placeholder",
    label: "Code placeholder",
    hideLabel: true,
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Code",
    value: "Code",
    label: "Code",
    hideLabel: true,
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Code-with-error",
    placeholder: "Code",
    label: "Code",
    hideLabel: true,
    appearance: "code",
    error: "Does not compute lorem ipsum dolor sit amet consecatur ",
    onChange: onChange
  }), _react["default"].createElement(_Input.Input, {
    id: "Code-with-error-and-label",
    placeholder: "Code",
    appearance: "code",
    orientation: "horizontal",
    error: "Does not compute lorem ipsum dolor sit amet consecatur ",
    onChange: onChange,
    label: "horizontal"
  }));
};

exports.code = code;
