// @storybook/design-system/dist/components/Link.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = Link;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _polished = require("polished");

var _Icon = require("./Icon");

var _styles = require("./shared/styles");

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", ";\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  /* reset button styles */\n  background: none;\n  color: inherit;\n  border: none;\n  padding: 0;\n  font: inherit;\n  cursor: pointer;\n  outline: inherit;\n\n  ", ";\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      > svg:last-of-type {\n        height: 0.7em;\n        width: 0.7em;\n        margin-right: 0;\n        margin-left: 0.25em;\n        bottom: auto;\n        vertical-align: inherit;\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      border: 0;\n      border-radius: 0;\n      background: none;\n      padding: 0;\n      font-size: inherit;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: inherit;\n\n      &:hover,\n      &:active {\n        color: inherit;\n        text-decoration: underline;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      svg {\n        height: 1em;\n        width: 1em;\n        vertical-align: middle;\n        position: relative;\n        bottom: 0;\n        margin-right: 0;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  transition: transform 150ms ease-out, color 150ms ease-out;\n  text-decoration: none;\n\n  color: ", ";\n\n  &:hover,\n  &:focus {\n    cursor: pointer;\n    transform: translateY(-1px);\n    color: ", ";\n  }\n  &:active {\n    transform: translateY(0);\n    color: ", ";\n  }\n\n  svg {\n    display: inline-block;\n    height: 1em;\n    width: 1em;\n    vertical-align: text-top;\n    position: relative;\n    bottom: -0.125em;\n    margin-right: 0.4em;\n  }\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var linkStyles = (0, _styledComponents.css)(_templateObject(), _styles.color.secondary, (0, _polished.darken)(0.07, _styles.color.secondary), (0, _polished.darken)(0.1, _styles.color.secondary), function (props) {
  return props.containsIcon && (0, _styledComponents.css)(_templateObject2());
}, function (props) {
  return props.secondary && (0, _styledComponents.css)(_templateObject3(), _styles.color.mediumdark, _styles.color.dark, _styles.color.darker);
}, function (props) {
  return props.tertiary && (0, _styledComponents.css)(_templateObject4(), _styles.color.dark, _styles.color.darkest, _styles.color.mediumdark);
}, function (props) {
  return props.nochrome && (0, _styledComponents.css)(_templateObject5());
}, function (props) {
  return props.inverse && (0, _styledComponents.css)(_templateObject6(), _styles.color.lightest, _styles.color.lighter, _styles.color.light);
}, function (props) {
  return props.isButton && (0, _styledComponents.css)(_templateObject7());
});

var LinkInner = _styledComponents["default"].span(_templateObject8(), function (props) {
  return props.withArrow && (0, _styledComponents.css)(_templateObject9());
});

var LinkA = _styledComponents["default"].a(_templateObject10(), linkStyles);

var LinkButton = _styledComponents["default"].button(_templateObject11(), linkStyles);

var applyStyle = function applyStyle(LinkWrapper) {
  return LinkWrapper && (0, _styledComponents["default"])(function (_ref) {
    var containsIcon = _ref.containsIcon,
        inverse = _ref.inverse,
        nochrome = _ref.nochrome,
        secondary = _ref.secondary,
        tertiary = _ref.tertiary,
        linkWrapperRest = (0, _objectWithoutProperties2["default"])(_ref, ["containsIcon", "inverse", "nochrome", "secondary", "tertiary"]);
    return _react["default"].createElement(LinkWrapper, linkWrapperRest);
  })(_templateObject12(), linkStyles);
};
/**
 * Links can contains text and/or icons. Be careful using only icons, you must provide a text alternative via aria-label for accessibility.
 */


function Link(_ref2) {
  var isButton = _ref2.isButton,
      withArrow = _ref2.withArrow,
      LinkWrapper = _ref2.LinkWrapper,
      children = _ref2.children,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["isButton", "withArrow", "LinkWrapper", "children"]);

  var content = _react["default"].createElement(_react.Fragment, null, _react["default"].createElement(LinkInner, {
    withArrow: withArrow
  }, children, withArrow && _react["default"].createElement(_Icon.Icon, {
    icon: "arrowright"
  })));

  var StyledLinkWrapper = applyStyle(LinkWrapper);
  var SelectedLink = LinkA;

  if (LinkWrapper) {
    SelectedLink = StyledLinkWrapper;
  } else if (isButton) {
    SelectedLink = LinkButton;
  }

  return _react["default"].createElement(SelectedLink, rest, content);
}

Link.propTypes = {
  isButton: _propTypes["default"].bool,
  children: _propTypes["default"].node,
  withArrow: _propTypes["default"].bool,
  containsIcon: _propTypes["default"].bool,
  LinkWrapper: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].object]),
  inverse: _propTypes["default"].bool,
  nochrome: _propTypes["default"].bool,
  secondary: _propTypes["default"].bool,
  tertiary: _propTypes["default"].bool
};
Link.defaultProps = {
  isButton: false,
  children: null,
  withArrow: false,
  containsIcon: false,
  LinkWrapper: undefined,
  inverse: false,
  nochrome: false,
  secondary: false,
  tertiary: false
};
