// @storybook/design-system/dist/components/Link.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.hasLinkWrapper = exports.isButton = exports.icon = exports.containsIcon = exports.withArrow = exports.all = exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _addonActions = require("@storybook/addon-actions");

var _Icon = require("./Icon");

var _Link = require("./Link");

var _StoryLinkWrapper = require("./StoryLinkWrapper");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  && {\n    color: red;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var CustomLink = (0, _styledComponents["default"])(_Link.Link)(_templateObject());
var onLinkClick = (0, _addonActions.action)('onLinkClick');
var _default = {
  title: 'Design System|Link',
  component: _Link.Link
};
exports["default"] = _default;

var all = function all() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_Link.Link, {
    href: "https://learnstorybook.com"
  }, "default"), ' ', _react["default"].createElement(_Link.Link, {
    secondary: true,
    href: "https://learnstorybook.com"
  }, "secondary"), ' ', _react["default"].createElement(_Link.Link, {
    tertiary: true,
    href: "https://learnstorybook.com"
  }, "tertiary"), ' ', _react["default"].createElement(_Link.Link, {
    nochrome: true,
    href: "https://learnstorybook.com"
  }, "nochrome"), ' ', _react["default"].createElement("span", {
    style: {
      background: '#333'
    }
  }, _react["default"].createElement(_Link.Link, {
    inverse: true,
    href: "https://learnstorybook.com"
  }, "inverse")));
};

exports.all = all;

var withArrow = function withArrow() {
  return _react["default"].createElement(_Link.Link, {
    containsIcon: true,
    withArrow: true,
    href: "https://learnstorybook.com"
  }, "withArrow shows an arrow behind the link");
};

exports.withArrow = withArrow;

var containsIcon = function containsIcon() {
  return _react["default"].createElement(_Link.Link, {
    containsIcon: true,
    href: "https://learnstorybook.com",
    "aria-label": "Toggle side bar"
  }, _react["default"].createElement(_Icon.Icon, {
    icon: "sidebar",
    "aria-hidden": true
  }));
};

exports.containsIcon = containsIcon;

var icon = function icon() {
  return _react["default"].createElement(_Link.Link, {
    href: "https://learnstorybook.com"
  }, _react["default"].createElement(_Icon.Icon, {
    icon: "discord",
    "aria-hidden": true
  }), "Link with an icon in front");
};

exports.icon = icon;

var isButton = function isButton() {
  return (
    /* eslint-disable-next-line */
    _react["default"].createElement(_Link.Link, {
      isButton: true,
      onClick: onLinkClick
    }, "is actually a button")
  );
};

exports.isButton = isButton;

var hasLinkWrapper = function hasLinkWrapper() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_Link.Link, {
    tertiary: true,
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    href: "http://storybook.js.org"
  }, "has a LinkWrapper like GatsbyLink or NextLink"), _react["default"].createElement("br", null), _react["default"].createElement(CustomLink, {
    tertiary: true,
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    href: "http://storybook.js.org"
  }, "has a LinkWrapper like GatsbyLink or NextLink with custom styling"));
};

exports.hasLinkWrapper = hasLinkWrapper;
