// @storybook/design-system/dist/components/ProgressDots.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.largeComplete = exports.complete = exports.halfway = exports.starting = exports.loading = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ProgressDots = require("./ProgressDots");

var _default = {
  title: 'Design System|ProgressDots',
  component: _ProgressDots.ProgressDots
};
exports["default"] = _default;

var loading = function loading() {
  return _react["default"].createElement(_ProgressDots.ProgressDots, {
    loading: true
  });
};

exports.loading = loading;

var starting = function starting() {
  return _react["default"].createElement(_ProgressDots.ProgressDots, {
    steps: 4,
    progress: 1
  });
};

exports.starting = starting;

var halfway = function halfway() {
  return _react["default"].createElement(_ProgressDots.ProgressDots, {
    steps: 4,
    progress: 2
  });
};

exports.halfway = halfway;

var complete = function complete() {
  return _react["default"].createElement(_ProgressDots.ProgressDots, {
    steps: 4,
    progress: 4
  });
};

exports.complete = complete;

var largeComplete = function largeComplete() {
  return _react["default"].createElement(_ProgressDots.ProgressDots, {
    steps: 4,
    progress: 4,
    size: "large"
  });
};

exports.largeComplete = largeComplete;
largeComplete.story = {
  name: 'large complete'
};
