// @storybook/design-system/dist/components/Radio.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.checked = exports.unchecked = exports.allRadios = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Radio = require("./Radio");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'Design System|forms/Radio',
  component: _Radio.Radio
};
exports["default"] = _default;

var allRadios = function allRadios() {
  return _react["default"].createElement("form", null, _react["default"].createElement(_Radio.Radio, {
    id: "Mice",
    label: "Mice",
    value: "mice",
    checked: true,
    onChange: onChange
  }), _react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    value: "dogs",
    onChange: onChange
  }), _react["default"].createElement(_Radio.Radio, {
    id: "Cats",
    label: "Cats",
    onChange: onChange,
    error: "There's a snake in my boots"
  }), _react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    description: "15 canines",
    value: "dogs",
    onChange: onChange
  }), _react["default"].createElement(_Radio.Radio, {
    id: "Mice",
    label: "Mice",
    value: "mice",
    onChange: onChange,
    checked: true,
    appearance: "secondary"
  }), _react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    value: "dogs",
    onChange: onChange,
    appearance: "secondary"
  }));
};

exports.allRadios = allRadios;
allRadios.story = {
  name: 'all radios'
};

var unchecked = function unchecked() {
  return _react["default"].createElement(_Radio.Radio, {
    id: "Mice",
    label: "Mice",
    hideLabel: true,
    value: "mice",
    onChange: onChange
  });
};

exports.unchecked = unchecked;

var checked = function checked() {
  return _react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    hideLabel: true,
    value: "dogs",
    checked: true,
    onChange: onChange
  });
};

exports.checked = checked;
