// @storybook/design-system/dist/components/Select.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tertiary = exports.secondary = exports.defaultStory = exports.allSelects = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Select = require("./Select");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'Design System|forms/Select',
  component: _Select.Select
};
exports["default"] = _default;

var allSelects = function allSelects() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Select.Select, {
    id: "Primary",
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }, {
      label: 'Mouse',
      value: 'value3'
    }],
    onChange: onChange,
    label: "Animal",
    hideLabel: true
  }), _react["default"].createElement(_Select.Select, {
    id: "Secondary",
    value: "value1",
    options: [{
      label: 'Secondary',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }, {
      label: 'Mouse',
      value: 'value3'
    }],
    appearance: "secondary",
    onChange: onChange,
    label: "Animal",
    hideLabel: true
  }), _react["default"].createElement(_Select.Select, {
    id: "Tertiary",
    value: "value1",
    options: [{
      label: 'Tertiary',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }, {
      label: 'Mouse',
      value: 'value3'
    }],
    appearance: "tertiary",
    onChange: onChange,
    label: "Animal",
    hideLabel: true
  }));
};

exports.allSelects = allSelects;
allSelects.story = {
  name: 'All selects'
};

var defaultStory = function defaultStory() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Select.Select, {
    id: "Primary",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Primary-disabled",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    disabled: true,
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Primary-with-iconn",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Primary-in-progress",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    onChange: onChange,
    inProgress: true
  }), _react["default"].createElement(_Select.Select, {
    id: "Primary-with-error",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    error: "There's a snake in my boots",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Primary-with-icon-and-error",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    error: "There's a snake in my boots",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Primary-with-label",
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    label: "Label",
    onChange: onChange
  }));
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};

var secondary = function secondary() {
  return _react["default"].createElement("form", {
    style: {
      background: '#FFFFFF',
      padding: '3em'
    }
  }, _react["default"].createElement(_Select.Select, {
    id: "Secondary",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Secondary-disabled",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    disabled: true,
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Secondary-with-icon",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Secondary-with-error",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    error: "There's a snake in my boots",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Secondary-with-icon-and-error",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    error: "There's a snake in my boots",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Secondary-with-label",
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chroma",
    label: "Label",
    appearance: "secondary",
    onChange: onChange
  }));
};

exports.secondary = secondary;

var tertiary = function tertiary() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Select.Select, {
    id: "Tertiary",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Select.Select, {
    id: "Tertiary-disabled",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    disabled: true,
    appearance: "tertiary",
    onChange: onChange
  }));
};

exports.tertiary = tertiary;
