// @storybook/design-system/dist/components/Spinner.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.inline = exports.inForm = exports.inverse = exports.defaultStory = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Spinner = require("./Spinner");

var _default = {
  title: 'Design System|Spinner',
  component: _Spinner.Spinner
};
exports["default"] = _default;

var defaultStory = function defaultStory() {
  return _react["default"].createElement("div", {
    style: {
      background: 'white',
      padding: '1.5em',
      position: 'relative'
    }
  }, _react["default"].createElement(_Spinner.Spinner, null));
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};

var inverse = function inverse() {
  return _react["default"].createElement("div", {
    style: {
      background: 'black',
      padding: '1.5em',
      position: 'relative'
    }
  }, _react["default"].createElement(_Spinner.Spinner, {
    inverse: true
  }));
};

exports.inverse = inverse;
inverse.story = {
  name: 'Inverse'
};

var inForm = function inForm() {
  return _react["default"].createElement("div", {
    style: {
      background: '#eee',
      padding: '1.5em',
      position: 'relative'
    }
  }, _react["default"].createElement(_Spinner.Spinner, {
    inForm: true
  }));
};

exports.inForm = inForm;

var inline = function inline() {
  return _react["default"].createElement("div", {
    style: {
      background: '#eee',
      padding: '1.5em',
      position: 'relative'
    }
  }, _react["default"].createElement(_Spinner.Spinner, {
    inline: true
  }));
};

exports.inline = inline;
