// @storybook/design-system/dist/components/Textarea.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.code = exports.tertiary = exports.secondary = exports.defaultStory = exports.allTextareas = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Textarea = require("./Textarea");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'Design System|forms/Textarea',
  component: _Textarea.Textarea
};
exports["default"] = _default;

var allTextareas = function allTextareas() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Textarea.Textarea, {
    id: "Default",
    label: "Default textarea",
    hideLabel: true,
    value: "Default",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Secondary",
    label: "Secondary textarea",
    hideLabel: true,
    value: "Secondary",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Tertiary",
    label: "Tertiary textarea",
    hideLabel: true,
    value: "Tertiary",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Code",
    label: "Code textarea",
    hideLabel: true,
    value: "Code",
    appearance: "code",
    onChange: onChange
  }));
};

exports.allTextareas = allTextareas;
allTextareas.story = {
  name: 'all Textareas'
};

var defaultStory = function defaultStory() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Textarea.Textarea, {
    id: "Placeholder",
    label: "Textarea with placeholder",
    hideLabel: true,
    value: "placeholder",
    placeholder: "Placeholder",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "With-value",
    label: "Textarea with value",
    hideLabel: true,
    value: "With value",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Disabled",
    label: "Disabled textarea",
    hideLabel: true,
    value: "Disabled",
    disabled: true,
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label",
    label: "Label",
    value: "Label",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Error",
    label: "Textarea with error",
    hideLabel: true,
    value: "Error",
    error: "There's a snake in my boots",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label-and-error",
    value: "Label and error",
    label: "Cats",
    error: "There's a snake in my boots",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Subtext",
    value: "Subtext",
    label: "Textarea with subtext",
    hideLabel: true,
    subtext: "140 chars left",
    onChange: onChange
  }));
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};

var secondary = function secondary() {
  return _react["default"].createElement("form", {
    style: {
      background: '#fff',
      padding: '3em'
    }
  }, _react["default"].createElement(_Textarea.Textarea, {
    id: "Placeholder",
    label: "Textarea with placeholder",
    hideLabel: true,
    value: "placeholder",
    placeholder: "Placeholder",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "With-value",
    label: "Textarea with value",
    hideLabel: true,
    value: "With value",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Disabled",
    label: "Disabled textarea",
    hideLabel: true,
    value: "Disabled",
    disabled: true,
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label",
    label: "Label",
    value: "Label",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Error",
    label: "Textarea with error",
    hideLabel: true,
    value: "Error",
    error: "There's a snake in my boots",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label-and-error",
    value: "Label and error",
    label: "Cats",
    error: "There's a snake in my boots",
    appearance: "secondary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Subtext",
    value: "Subtext",
    label: "Textarea with subtext",
    hideLabel: true,
    subtext: "140 chars left",
    appearance: "secondary",
    onChange: onChange
  }));
};

exports.secondary = secondary;

var tertiary = function tertiary() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Textarea.Textarea, {
    id: "Placeholder",
    label: "Textarea with placeholder",
    hideLabel: true,
    value: "placeholder",
    placeholder: "Placeholder",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "With-value",
    label: "Textarea with value",
    hideLabel: true,
    value: "With value",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Disabled",
    label: "Disabled textarea",
    hideLabel: true,
    value: "Disabled",
    disabled: true,
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label",
    label: "Label",
    value: "Label",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Error",
    label: "Textarea with error",
    hideLabel: true,
    value: "Error",
    error: "There's a snake in my boots",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label-and-error",
    value: "Label and error",
    label: "Cats",
    error: "There's a snake in my boots",
    appearance: "tertiary",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Subtext",
    value: "Subtext",
    label: "Textarea with subtext",
    hideLabel: true,
    subtext: "140 chars left",
    appearance: "tertiary",
    onChange: onChange
  }));
};

exports.tertiary = tertiary;

var code = function code() {
  return _react["default"].createElement("form", {
    style: {
      background: '#EEEEEE',
      padding: '3em'
    }
  }, _react["default"].createElement(_Textarea.Textarea, {
    id: "Placeholder",
    label: "Textarea with placeholder",
    hideLabel: true,
    value: "placeholder",
    placeholder: "Code placeholder",
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "With-value",
    label: "Textarea with value",
    hideLabel: true,
    value: "Code",
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Error",
    label: "Textarea with error",
    hideLabel: true,
    value: "Code",
    appearance: "code",
    error: "Does not compute",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label",
    label: "Label",
    value: "Label",
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label-and-error",
    value: "Label and error",
    label: "Cats",
    error: "There's a snake in my boots",
    appearance: "code",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label-horizontal",
    value: "Label",
    label: "Cats",
    appearance: "code",
    orientation: "horizontal",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Label-and-error-horizontal",
    value: "Label and error",
    label: "Cats",
    error: "There's a snake in my boots",
    appearance: "code",
    orientation: "horizontal",
    onChange: onChange
  }), _react["default"].createElement(_Textarea.Textarea, {
    id: "Subtext",
    value: "Subtext",
    label: "Textarea with subtext",
    hideLabel: true,
    subtext: "140 chars left",
    appearance: "code",
    onChange: onChange
  }));
};

exports.code = code;
