// @storybook/design-system/dist/components/modal/Modal.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.notVisible = exports.visibleTall = exports.visibleShort = exports["default"] = exports.actions = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Modal = require("./Modal");

var _Button = require("../Button");

var actions = {
  onClose: (0, _addonActions.action)('onClose')
};
exports.actions = actions;
var _default = {
  title: 'Design System|modal/Modal',
  component: _Modal.Modal,
  decorators: [function (storyFn) {
    return _react["default"].createElement("div", {
      style: {
        width: '1200px',
        height: '800px',
        background: 'purple',
        color: 'white'
      }
    }, "This is an example background ", storyFn());
  }],
  excludeStories: ['actions']
};
exports["default"] = _default;

var visibleShort = function visibleShort() {
  return _react["default"].createElement(_Modal.Modal, (0, _extends2["default"])({
    isOpen: true
  }, actions), function (_ref) {
    var onClose = _ref.onClose;
    return _react["default"].createElement("div", {
      style: {
        background: 'papayawhip',
        fontSize: '12px'
      }
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent egestas, ante eu ornare tincidunt, justo velit ultrices est, eu porta libero ante malesuada ante. Maecenas fermentum ex nec justo tempor rutrum. Donec sapien nisl, sollicitudin sit amet lorem eget, congue pharetra nibh. Nulla eu lobortis ante. Vivamus tortor sapien, dictum sit amet lacus sit amet, cursus porttitor sapien. Donec a massa et felis dictum eleifend. Etiam in dui at magna imperdiet gravida. Nam id tortor in leo facilisis elementum sit amet accumsan nunc. Maecenas lacus metus, dapibus aliquam mauris ac, luctus gravida ex.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent egestas, ante eu ornare tincidunt, justo velit ultrices est, eu porta libero ante malesuada ante. Maecenas fermentum ex nec justo tempor rutrum. Donec sapien nisl, sollicitudin sit amet lorem eget, congue pharetra nibh. Nulla eu lobortis ante. Vivamus tortor sapien, dictum sit amet lacus sit amet, cursus porttitor sapien. Donec a massa et felis dictum eleifend. Etiam in dui at magna imperdiet gravida. Nam id tortor in leo facilisis elementum sit amet accumsan nunc. Maecenas lacus metus, dapibus aliquam mauris ac, luctus gravida ex.", ' ', _react["default"].createElement(_Button.Button, {
      appearance: "primary",
      onClick: onClose,
      role: "button",
      tabIndex: "0"
    }, "Close"));
  });
};

exports.visibleShort = visibleShort;
visibleShort.story = {
  name: 'visible short'
};

var visibleTall = function visibleTall() {
  return _react["default"].createElement(_Modal.Modal, (0, _extends2["default"])({
    isOpen: true
  }, actions), function (_ref2) {
    var onClose = _ref2.onClose;
    return _react["default"].createElement("div", {
      style: {
        background: 'papayawhip',
        fontSize: '36px'
      }
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent egestas, ante eu ornare tincidunt, justo velit ultrices est, eu porta libero ante malesuada ante. Maecenas fermentum ex nec justo tempor rutrum. Donec sapien nisl, sollicitudin sit amet lorem eget, congue pharetra nibh. Nulla eu lobortis ante. Vivamus tortor sapien, dictum sit amet lacus sit amet, cursus porttitor sapien. Donec a massa et felis dictum eleifend. Etiam in dui at magna imperdiet gravida. Nam id tortor in leo facilisis elementum sit amet accumsan nunc. Maecenas lacus metus, dapibus aliquam mauris ac, luctus gravida ex.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent egestas, ante eu ornare tincidunt, justo velit ultrices est, eu porta libero ante malesuada ante. Maecenas fermentum ex nec justo tempor rutrum. Donec sapien nisl, sollicitudin sit amet lorem eget, congue pharetra nibh. Nulla eu lobortis ante. Vivamus tortor sapien, dictum sit amet lacus sit amet, cursus porttitor sapien. Donec a massa et felis dictum eleifend. Etiam in dui at magna imperdiet gravida. Nam id tortor in leo facilisis elementum sit amet accumsan nunc. Maecenas lacus metus, dapibus aliquam mauris ac, luctus gravida ex.", ' ', _react["default"].createElement(_Button.Button, {
      appearance: "primary",
      onClick: onClose,
      role: "button",
      tabIndex: "0"
    }, "Close"));
  });
};

exports.visibleTall = visibleTall;
visibleTall.story = {
  name: 'visible tall'
};

var notVisible = function notVisible() {
  return _react["default"].createElement(_Modal.Modal, (0, _extends2["default"])({
    isOpen: false
  }, actions), function (_ref3) {
    var onClose = _ref3.onClose;
    return _react["default"].createElement("div", null, _react["default"].createElement(_Button.Button, {
      appearance: "primary",
      onClick: onClose,
      role: "button",
      tabIndex: "0"
    }, "Close"));
  });
};

exports.notVisible = notVisible;
notVisible.story = {
  name: 'not visible'
};
