// @storybook/design-system/dist/components/modal/WithModal.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.startsOpen = exports.startsClosed = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Button = require("../Button");

var _WithModal = _interopRequireDefault(require("./WithModal"));

// eslint-disable-next-line react/prop-types
var ModalContents = function ModalContents(_ref) {
  var onClose = _ref.onClose;
  return _react["default"].createElement("div", {
    style: {
      background: 'papayawhip',
      fontSize: '12px'
    }
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent egestas, ante eu ornare tincidunt, justo velit ultrices est, eu porta libero ante malesuada ante. Maecenas fermentum ex nec justo tempor rutrum. Donec sapien nisl, sollicitudin sit amet lorem eget, congue pharetra nibh. Nulla eu lobortis ante. Vivamus tortor sapien, dictum sit amet lacus sit amet, cursus porttitor sapien. Donec a massa et felis dictum eleifend. Etiam in dui at magna imperdiet gravida. Nam id tortor in leo facilisis elementum sit amet accumsan nunc. Maecenas lacus metus, dapibus aliquam mauris ac, luctus gravida ex.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent egestas, ante eu ornare tincidunt, justo velit ultrices est, eu porta libero ante malesuada ante. Maecenas fermentum ex nec justo tempor rutrum. Donec sapien nisl, sollicitudin sit amet lorem eget, congue pharetra nibh. Nulla eu lobortis ante. Vivamus tortor sapien, dictum sit amet lacus sit amet, cursus porttitor sapien. Donec a massa et felis dictum eleifend. Etiam in dui at magna imperdiet gravida. Nam id tortor in leo facilisis elementum sit amet accumsan nunc. Maecenas lacus metus, dapibus aliquam mauris ac, luctus gravida ex.", ' ', _react["default"].createElement(_Button.Button, {
    appearance: "primary",
    onClick: onClose,
    role: "button",
    tabIndex: "0"
  }, "Close"));
};

var _default = {
  title: 'Design System|modal/WithModal',
  decorators: [function (storyFn) {
    return _react["default"].createElement("div", {
      style: {
        width: '1200px',
        height: '800px',
        background: 'purple',
        color: 'white'
      }
    }, "This is an example background ", storyFn());
  }],
  parameters: {
    component: _WithModal["default"]
  }
};
exports["default"] = _default;

var startsClosed = function startsClosed() {
  return _react["default"].createElement(_WithModal["default"], {
    modal: ModalContents
  }, function (_ref2) {
    var onOpen = _ref2.onOpen;
    return _react["default"].createElement("div", null, _react["default"].createElement(_Button.Button, {
      appearance: "primary",
      onClick: onOpen,
      role: "button",
      tabIndex: "0"
    }, "Open Modal"));
  });
};

exports.startsClosed = startsClosed;
startsClosed.story = {
  name: 'starts closed'
};

var startsOpen = function startsOpen() {
  return _react["default"].createElement(_WithModal["default"], {
    startOpen: true,
    modal: ModalContents
  }, function (_ref3) {
    var onOpen = _ref3.onOpen;
    return _react["default"].createElement("div", null, _react["default"].createElement(_Button.Button, {
      appearance: "primary",
      onClick: onOpen,
      role: "button",
      tabIndex: "0"
    }, "Open Modal"));
  });
};

exports.startsOpen = startsOpen;
startsOpen.story = {
  name: 'starts open'
};
