// @storybook/design-system/dist/components/modal/WithModal.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var _recompose = require("recompose");

var _Modal = require("./Modal");

function WithModal(_ref) {
  var isOpen = _ref.isOpen,
      onOpen = _ref.onOpen,
      onClose = _ref.onClose,
      modal = _ref.modal,
      children = _ref.children;
  return _react["default"].createElement(_react.Fragment, null, _react.Children.only(children({
    isOpen: isOpen,
    onOpen: onOpen,
    onClose: onClose
  })), _react["default"].createElement(_Modal.Modal, {
    isOpen: isOpen,
    onClose: onClose
  }, modal));
}

WithModal.propTypes = {
  isOpen: _propTypes.PropTypes.bool.isRequired,
  onOpen: _propTypes.PropTypes.func.isRequired,
  onClose: _propTypes.PropTypes.func.isRequired,
  modal: _propTypes.PropTypes.func.isRequired,
  children: _propTypes.PropTypes.func.isRequired
};

var _default = (0, _recompose.compose)((0, _recompose.withState)('isOpen', 'onSetIsOpen', function (_ref2) {
  var startOpen = _ref2.startOpen;
  return !!startOpen;
}), (0, _recompose.withHandlers)({
  onOpen: function onOpen(_ref3) {
    var onSetIsOpen = _ref3.onSetIsOpen;
    return function () {
      return onSetIsOpen(true);
    };
  },
  onClose: function onClose(_ref4) {
    var onSetIsOpen = _ref4.onSetIsOpen,
        _onClose = _ref4.onClose;
    return function () {
      onSetIsOpen(false);

      if (_onClose) {
        _onClose();
      }
    };
  }
}))(WithModal);

exports["default"] = _default;
